src/object.js

```diff
@@ -41,28 +41,49 @@ function TransisObject() {
 
 TransisObject.displayName = 'Transis.Object';
 
-// Internal: Processes a property change by traversing the property dependency graph and forwarding
-// changes to proxy objects.
-function didChange(object, name) {
-  (object.__changedProps__ = object.__changedProps__ || {})[name] = true;
-  changedObjects[object.objectId] = object;
-  uncache.call(object, name);
+// Internal: Processes recorded property changes by traversing the property dependency graph and
+// forwarding changes to proxy objects.
+function processChangedObjects() {
+  let seen = {};
+  let head;
 
-  if (object.__deps__) {
-    let deps = object.__deps__[name];
+  for (let id in changedObjects) {
+    for (let k in changedObjects[id].__changedProps__) {
+      head = {object: changedObjects[id], name: k, next: head};
+      seen[changedObjects[id].objectId + k] = true;
+    }
+  }
+
+  while (head) {
+    let object = head.object;
+    let name = head.name;
+    let deps;
+
+    head = head.next;
 
-    if (deps) {
+    (object.__changedProps__ = object.__changedProps__ || {})[name] = true;
+    changedObjects[object.objectId] = object;
+    if (object.__cache__) { delete object.__cache__[name]; }
+
+    if ((deps = object.__deps__ && object.__deps__[name])) {
       for (let i = 0, n = deps.length; i < n; i++) {
-        didChange(object, deps[i]);
+        if (!seen[object.objectId + deps[i]]) {
```

Comment:
Could we store the "seen key" in a variable?  Also for the `proxyObject.objectId + proxyName` below too.
I worry the repetition will be error prone to maintain (update one but forget the other) and probably a slight perf hit (additional lookups on the VM and additional string concatenations)